TFNetworkLayer.py

```diff
@@ -3345,12 +3345,14 @@ def __init__(self, axis, dim=1, **kwargs):
     data = self.input_data
     if isinstance(axis, int):
       data = data.copy_as_batch_major()
-    axis = self._get_axis(data=data, axis=axis)
+    axis_index = self._get_axis(data=data, axis=axis)
     from TFUtil import expand_dims_unbroadcast
-    self.output.placeholder = expand_dims_unbroadcast(data.placeholder, axis=axis, dim=dim)
+    self.output.placeholder = expand_dims_unbroadcast(data.placeholder, axis=axis_index, dim=dim)
     self.output.size_placeholder = {
-      (i if (data.get_batch_axis(i) < axis) else i + 1): v
+      (i if (data.get_batch_axis(i) < axis_index) else i + 1): v
       for (i, v) in data.size_placeholder.items()}
+    if axis.lower() in ["spatial", "time", "t"] and self.output.time_dim_axis is None:
+        self.output.time_dim_axis = axis_index
```

Comment:
The indentation here is too much.

Reply:
Actually the logic to set `time_dim_axis` should better also be in `get_out_data_from_opts`, and not in `__init__`.

Reply:
Ok, I'll change that tomorrow.